oe_content.module

```diff
@@ -2,9 +2,51 @@
 
 /**
  * @file
- * OpenEuropa content module.
+ * The OE Content module.
  */
 
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\rdf_entity\Entity\Rdf;
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function oe_content_entity_base_field_info(EntityTypeInterface $entity_type) {
+  // Add the "provenance_uri" field to the RDF entity type.
+  if ($entity_type->id() !== 'rdf_entity') {
+    return;
+  }
+
+  $fields = [];
+
+  $fields['provenance_uri'] = BaseFieldDefinition::create('uri')
+    ->setLabel(t('Provenance URI'))
+    ->setDescription(t('A URI that indicates the provenance of the entity.'))
+    ->setTranslatable(FALSE);
+
+  return $fields;
+}
+
+/**
+ * Implements hook_entity_presave().
+ */
+function oe_content_entity_presave(EntityInterface $entity) {
+  // RDF entities need to be saved with a provenance URI.
+  if (!$entity instanceof Rdf) {
+    return;
+  }
+
+  $uri = \Drupal::config('oe_content.settings')->get('provenance_uri');
+  if (!$uri) {
```

Comment:
Let's throw a critical exception here. The website is unusable if the provenance uri is not set. Theoretically it should go earlier on, like when accessing the entity create/edit form, but it's ok also here.